draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx

```diff
@@ -15,6 +15,16 @@ import NavigationTab, { NavigationTabProps } from "./NavigationTabs"
 import SecondaryActions from "./SecondaryActions"
 import styles from "./TitleBlockZen.scss"
 
+/*
+  This type exists to support omitting keys from a union or intersection type in a distributive manner. `Omit` out of the box will cause you to lose any union or intersection information about the type,
+  thus you might lose the ability to access some fields that only exist conditionally based on another (e.g. discriminated unions).
+  `T extends any ? Omit<T, K>` is a trick used to spread the action of Omit across every variant within the union or intersection.
+  So, if T is something like `{foo: string} | {bar: string}`, it becomes `Omit<{foo: string}, K> | Omit<{bar: string}, K>
+*/
```

Comment:
This is super interesting! I wanted to read more and found this post really thorough: https://davidgomes.com/pick-omit-over-union-types-in-typescript/

Perhaps we could include it for those that find their way here?

```suggestion
/*
  This type exists to support omitting keys from a union or intersection type in a distributive manner. `Omit` out of the box will cause you to lose any union or intersection information about the type,
  thus you might lose the ability to access some fields that only exist conditionally based on another (e.g. discriminated unions).
  `T extends any ? Omit<T, K>` is a trick used to spread the action of Omit across every variant within the union or intersection.
  So, if T is something like `{foo: string} | {bar: string}`, it becomes `Omit<{foo: string}, K> | Omit<{bar: string}, K>
  https://davidgomes.com/pick-omit-over-union-types-in-typescript/
*/
```

Reply:
Yeah good idea :) thanks!